crowdsourcer/management/commands/import_foi_data.py

```diff
@@ -0,0 +1,323 @@
+            "private_notes": notes,
+            "council": row["public_body_name"],
+        }
+
+        q_details = self.sheet_map[name]
+        if pd.isna(row.iloc[7]):
+            return None
+        try:
+            value = int(row.iloc[7])
+        except ValueError:
+            self.warnings.append(f"bad data in row: {row.iloc[7]}")
+            return None
+
+        if q_details["type"] == "yes_no":
+            if value == 1:
+                defaults["option"] = self.get_option_for_question(q, "Yes")
+            elif value == 0:
+                defaults["option"] = self.get_option_for_question(q, "No")
+        elif q_details["type"] == "tiered":
+            if name == "B&H Q2":
+                defaults["multi_option"] = self.get_bh_2_answer(q, row, value)
+            elif name == "B&H Q3":
+                defaults["option"] = self.get_bh_3_answer(q, row, value)
+        else:
+            return None
+
+        return defaults
+
+    def create_options(self):
+        for sheet, details in self.sheet_map.items():
+
+            section = Section.objects.get(title=details["section"])
+            q = Question.objects.get(section=section, number=details["question"])
+
+            options = [
+                {"description": "Yes", "score": 1},
+                {"description": "No", "score": 0},
+                {"description": "No response", "score": 0},
+            ]
+
+            if details["type"] == "tiered":
+                options = self.tiered_options[sheet]["options"]
+                if q.question_type != "multiple_choice":
+                    q.question_type = "multiple_choice"
+                    q.save()
+
+            for option in options:
+                try:
+                    Option.objects.get_or_create(
+                        question=q,
+                        description=option["description"],
+                        defaults={
+                            "score": option["score"],
+                        },
+                    )
+                except Option.MultipleObjectsReturned:
+                    print(q, option["description"])
+                    continue
+
+    def handle(self, *args, **options):
+        self.create_options()
+
+        rt = ResponseType.objects.get(type="First Mark")
+        u, _ = User.objects.get_or_create(
+            username="FOI_importer",
+        )
+
+        council_lookup = self.get_council_lookup()
+
+        for name, details in self.sheet_map.items():
+            self.warnings = []
+            df = pd.read_excel(
+                self.foi_file,
+                sheet_name=name,
+                header=0,
+            )
+
+            df = df.dropna(axis="index", how="all")
+
+            section = Section.objects.get(title=details["section"])
+
+            q = Question.objects.get(
+                section=section,
+                number=details["question"],
+            )
+
+            if q.how_marked != "foi":
+                continue
```

Comment:
What causes this to happen? Question is listed as an FOI question above but not in the dataset.

It if happens at all, a comment on why it's ok, if never, would assert it never happens.